src/utils/userLocation.ts

```diff
@@ -0,0 +1,83 @@
+import firebase, { UserInfo } from "firebase/app";
+
+import { AnyRoom } from "types/rooms";
+
+import { updateUserProfile } from "pages/Account/helpers";
+import { useInterval } from "hooks/useInterval";
+
+import { getCurrentTimeInMilliseconds } from "./time";
+import { openRoomUrl, openUrl, venueInsideUrl } from "./url";
+
+const LOCATION_INCREMENT_SECONDS = 10;
+const LOCATION_INCREMENT_MS = LOCATION_INCREMENT_SECONDS * 1000;
+
+type LocationData = Record<string, number>;
+
+export const updateLocationData = (
+  userId: string,
+  newLocationData: LocationData = {}
+) => {
+  updateUserProfile(userId, {
+    lastSeenAt: getCurrentTimeInMilliseconds(),
+    lastSeenIn: newLocationData,
```

Comment:
This changes the logic from how this function used to work. Previously it had a bunch of logic to ensure it didn't remove data from this field that it shouldn't be updating, whereas now we are just wiping out the field entirely each time. While this may be the desired effect, it would be good to know your thought process, and why this way is better than what was there before.

Reply:
The reasoning behind this approach is that a person can be only in one location at any point in time. I was also thinking about removing `lastSeenAt` field, since `lastSeenIn`'s value is a dupe of `lastSeenAt` value. WDYT of it?

Reply:
I don't believe we currently enforce them being only in a single location though, if they have 2 tabs open, etc. So i'm not sure that "can only be in one location at any point of time" is true right now (even though I know you have thoughts/plans to change that)

---

We can't remove `lastSeenAt` currently because it persists, whereas `lastSeenIn` gets wiped out whenever room changes/they leave the platform/etc.

If you want to remove it as part of a greater change to how `lastSeenIn` works in general (so that it doesn't ever lose that data), I am open to exploring it as part of that work, but I expect it may have a bunch of related code changes that will need to be taken into consideration as part of it.